### ISSUE TITLE
Laravel Model structure

### ISSUE THREAD

**Issue by jannescb** [aw-studio/guidelines#25]
I was wondering if you follow a strict path when it comes to structuring your model classes.

```php
class Foo
{
    // traits
    // constants
    // variables
    // relations
    // scopes
    // accessors & mutators
    // methods
    ...
}

**Comment by cbl** [comment]
Sounds good to structure the order. I am in favor of defining accessors and getters first, since they extend attributes, and then relationships and scopes, which describe query methods.

**Comment by lpheller** [comment]
Yes, would be a nice thing to commit to a fixed structure. I just looked through a few files of mine. I don't think I'm very set in the order so far and basically only try to group by "type". So this actually couldn't hurt 😇

I think I would like it this way:
// traits
// variables
// (relationships)
// accessor / mutators
// scopes
// methods

Not quite sure about relationships yet, but accessors / mutators, scopes, and methods are the things that make the class grow the fastest, so I would want to place them further down.

**Comment by cbl** [comment]
@lpheller I think relationships and scopes should be placed together as they both concern eloquent queries.

**Comment by lpheller** [comment]
@cbl I got it, good point.

In Summary this would be:

// traits

// **Attribute related**
// constants
// variables
// accessor / mutators

// **Database / eloquent related**
// relationships
// scopes

// **General**
// methods


What about a static boot/booting method? Those are commonly placed at the top / below the variables?

**Comment by cbl** [comment]
@lpheller I think booting methods should be placed directly after variables as they construct the model in a way.

**Comment by lpheller** [comment]
@cbl before or after the accessors / mutators?

**Comment by cbl** [comment]
before, the first method MUST be the constructor and booting methods follow the constructor.

**Comment by lpheller** [comment]
Which brings us to:
```php 
class Foo extends Model {
    // traits
    // constants
    // variables
    // constructor
    // static boot / booting
    // accessors
    // mutators
    // relationships
    // scopes
    // methods
}
```

Anything else to consider? @jannescb what do you think?
